src/snap/utils.ts

```diff
@@ -17,3 +18,55 @@ export function isMetaMaskErrorObject(e: unknown): boolean {
   if (!("originalError" in e["data"])) return false;
   return true;
 }
+
+export function isElementVisible(
+  page: DappeteerPage,
+  selector: string,
+  timeout = 1000
+): Promise<boolean> {
+  return new Promise((resolve) => {
+    page
+      .waitForSelector(selector, { visible: true, timeout })
+      .then(() => {
+        resolve(true);
+      })
+      .catch(() => {
+        resolve(false);
+      });
+  });
+}
+
+function getWaitingPromise(
+  page: DappeteerPage,
+  selectorOrXpath: string,
+  timeout: number
+): Promise<DappeteerElementHandle<unknown, HTMLElement>> {
+  if (selectorOrXpath.startsWith("//")) {
+    return page.waitForXPath(selectorOrXpath, { timeout });
+  } else {
+    return page.waitForSelector(selectorOrXpath, { timeout });
+  }
+}
+
+interface IsFirstElementAppearsFirstParams {
+  selectorOrXpath1: string;
+  selectorOrXpath2: string;
+  timeout?: number;
+  page: DappeteerPage;
+}
+
+export async function isFirstElementAppearsFirst({
+  selectorOrXpath1,
+  selectorOrXpath2,
+  page,
+  timeout = 2000,
+}: IsFirstElementAppearsFirstParams): Promise<boolean> {
+  const promise1 = getWaitingPromise(page, selectorOrXpath1, timeout).then(
+    () => true
+  );
+  const promise2 = getWaitingPromise(page, selectorOrXpath2, timeout).then(
+    () => false
+  );
+
+  return await Promise.race([promise1, promise2]);
+}
```

Comment:
and with that approach you can simply iterate over a map

```suggestion
}: IsFirstElementAppearsFirstParams): Promise<boolean> {
  return await Promise.race(selectors.map(selector => getWaitingPromise(page, selector, timeout).then(
    () => false
  ));
}
```

Reply:
I'm not sure I can follow you here. Afaiu your suggestion will always return `false` for everything.
Having only 2 selectors is all that is needed in my case because I want to know whether or not the first or the second selector is found first. Hence it'll return `true` if the first selector is found first, and `false` if the second is found first. I don't really see the need or the use case for having more.

Reply:
Oh my bad I misread the function